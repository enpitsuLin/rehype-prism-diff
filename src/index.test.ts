import { describe, test, expect } from 'vitest';
import { rehype } from 'rehype';
import dedent from 'dedent';
import rehypePrismDiff from '.';
import rehypePrism, { RehypePrismOptions } from 'rehype-prism';
import { visit } from 'unist-util-visit';
import type { Element } from 'hast';

const addMeta = (meta?: string) => {
  if (!meta) return;
  return (tree: Element) => {
    visit(tree, 'element', (node, index, parent) => {
      if (node.tagName === 'code') {
        node.data = { meta: meta };
      }
    });
  };
};

describe('rehypre-prism plugin', () => {
  const processHtml = (html: string, setting?: { options?: RehypePrismOptions; meta?: string }) => {
    return rehype()
      .data('settings', { fragment: true })
      .use(addMeta, setting?.meta || '')
      .use(rehypePrism, setting?.options)
      .use(rehypePrismDiff)
      .processSync(html)
      .toString();
  };
  test('should add diff', () => {
    const meta = 'diff';
    const result = processHtml(
      dedent`
    <div>
      <pre>
      <code class="language-css">p { color: red }</code>
      </pre>
    </div>
    `,
      {
        meta
      }
    );
    expect(result).toMatchSnapshot();
  });
});
