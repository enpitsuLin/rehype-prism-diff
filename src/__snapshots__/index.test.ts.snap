// Vitest Snapshot v1

exports[`rehypre-prism plugin > should add diff 1`] = `
"<div>
  <pre class=\\"language-css\\">  <code class=\\"language-css code-diff\\"><span class=\\"token selector\\">p</span> <span class=\\"token punctuation\\">{</span> <span class=\\"token property\\">color</span><span class=\\"token punctuation\\">:</span> red <span class=\\"token punctuation\\">}</span></code>
  </pre>
</div>"
`;
